### DISCUSSION TITLE
Problem with Cookie library

### DISCUSSION THREAD

**Discussion by AlfWo** [fastapi/fastapi#6035] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [ ] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [ ] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [ ] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [ ] I commit to help with one of those options 👆

### Example Code

```python
from typing import Union
from fastapi import Cookie, FastAPI

app = FastAPI()


@app.get("/items/")
async def read_items(ads_id: Union[str, None] = Cookie(default=None)):
    return {"ads_id": ads_id}
```

### Description

Hi,

I'm new to fastApi, and I was following the web guide at:
https://fastapi.tiangolo.com/tutorial/cookie-params/

I have some trouble with the uploaded code.
	
No matter what I pass to as ads_id, I always get a null value in return.

I have used both swaggers and postman to try this API,and I'm using Python version 3.11.2,and fast API version 0.92.0

Does anyone know how to solve this?

### Operating System

Windows

### FastAPI Version

0.92.0

### Python Version

3.11.2

**Comment by mr-st0rm** [comment]
you can try get cookies from reqest object

**Comment by YuriiMotov** [comment, marked as answer]
For security reasons you can't send cookie using input field in Swagger interface.

The workaround is to create the endpoint that will set cookie:
```python
@app.get("/set_cookie/")
async def set_cookie(name: str, value: str, response: Response):
    response.set_cookie(key=name, value=value)
    return {"status": "Ok"}
```

Then before sending request to `/items/` use that endpoint to set the value of the cookie.

```
curl -X 'GET' 'http://localhost:8000/set_cookie/?name=ads_id&value=something' -H 'accept: application/json'
{"status": "Ok"}

curl -X 'GET' 'http://localhost:8000/items/' -H 'accept: application/json'
{"ads_id": "something"}

```

Full code example in details:

<details>

```python
from typing import Union
from fastapi import Cookie, FastAPI, Response

app = FastAPI()


@app.get("/items/")
async def read_items(ads_id: Union[str, None] = Cookie(default=None)):
    return {"ads_id": ads_id}


@app.get("/set_cookie/")
async def set_cookie(name: str, value: str, response: Response):
    response.set_cookie(key=name, value=value)
    return {"status": "Ok"}

```
</details>
